### PR TITLE
Added problems for C9, C11

### PR DESCRIPTION
**Pull request by Grizzlly**
For C9:

- 02-max_struct.c

For C11:

- 01-copy_text.c
- 02-copy_binary.c
- 04-append_text.c

Added .gitignore to ignore Visual Studio files.

### CODE REVIEW

File: Curs09/C/02-max_struct.c [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+#define NAMES_COUNT 5
+#define MAX_NAME_LENGTH 8
+
+struct Elev
+{
+	char Nume[MAX_NAME_LENGTH];
+	float Nota;
+};
+
+struct Elev* getRandomStudents(const char MulteNume[][MAX_NAME_LENGTH])
```
Comment (RaduNichita) [inline Curs09/C/02-max_struct.c:21]: Please use snake_case instead of camelCase.

Comment (RaduNichita) [inline Curs09/C/02-max_struct.c:21]:
```suggestion
struct Elev* getRandomStudents(const char MulteNume[][MAX_NAME_LENGTH])
```

change MulteNume to something more appropriate. (e.g: students_names).

File: Curs09/C/02-max_struct.c [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+{
+	struct Elev* elevi = malloc(sizeof(elevi) * NAMES_COUNT);
```
Comment (RaduNichita) [inline Curs09/C/02-max_struct.c:23]: Please name all variable in english.

Comment (RaduNichita) [inline Curs09/C/02-max_struct.c:23]: Please check malloc return value.

File: Curs09/C/02-max_struct.c [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+		if (maxNota.Nota < elevi[i].Nota)
+		{
+			maxNota = elevi[i];
+		}
+
+	}
+
+	printf("\nStudentul cu cea mai mare nota: %s\n", maxNota.Nume);
+
```
Comment (RaduNichita) [inline Curs09/C/02-max_struct.c:57]:
```suggestion
```

File: Curs09/C/02-max_struct.c [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+	free(elevi);
+
```
Comment (RaduNichita) [inline Curs09/C/02-max_struct.c:59]:
```suggestion
```

File: Curs09/C/02-max_struct.c [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+	return 0;
+}
+
+/*
+
+	compile: gcc 02-max_struct.c -o max_struct
```
Comment (RaduNichita) [inline Curs09/C/02-max_struct.c:65]: Please add the Makefile for this.

File: Curs11/C/01-copy_text.c [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+	Program care realizeaza copierea unui fisier text.
+
+*/
+
+#include <stdio.h>
+
+int main()
+{
+	FILE *copyFrom, *copyTo;
+	char filename[101];
```
Comment (RaduNichita) [inline Curs11/C/01-copy_text.c:12]: Don't use hardcoded value. make defines for them

File: Curs11/C/01-copy_text.c [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+	int read = 0;
+
+	printf("File to copy: ");
+	scanf("%100s", filename);
+
+	if ((copyFrom = fopen(filename, "r")) == NULL)
```
Comment (RaduNichita) [inline Curs11/C/01-copy_text.c:18]: do no open file in if statements.

File: Curs11/C/01-copy_text.c [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+#include <stdio.h>
+
+int main()
+{
+	FILE *copyFrom, *copyTo;
+	char filename[101];
+	int read = 0;
+
+	printf("File to copy: ");
+	scanf("%100s", filename);
```
Comment (RaduNichita) [inline Curs11/C/01-copy_text.c:16]: please use fgets for that instead.

File: Curs11/C/01-copy_text.c [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+	if ((copyFrom = fopen(filename, "r")) == NULL)
+	{
+		fprintf(stderr, "%s", "No such file! Exiting...");
```
Comment (RaduNichita) [inline Curs11/C/01-copy_text.c:20]: add newline

File: Curs11/C/01-copy_text.c [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+	
+	if ((copyTo = fopen(filename, "w")) == NULL)
+	{
+		fprintf(stderr, "%s", "Cannot open/create %s file! Exiting...",
+			filename);
+		return -1;
+	}
+
+	// Go character by character
+	read = fgetc(copyFrom);
```
Comment (RaduNichita) [inline Curs11/C/01-copy_text.c:35]: create a function to copy the files.

File: Curs11/C/01-copy_text.c [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+
+	compile: gcc 01-copy_text.c -o copy_text
+	run: ./copy_text
+
+	Output:
+
+		File to copy: {01-copy_text.c}
+		Copy to: {test.c}
+
+*/
```
Comment (RaduNichita) [inline Curs11/C/01-copy_text.c:59]: Add newline at end of the file

File: Curs11/C/01-copy_text.c [hunk 12]
```diff
@@ -0,0 +1,59 @@
...
+}
+
+/*
+
+	compile: gcc 01-copy_text.c -o copy_text
+	run: ./copy_text
+
+	Output:
+
+		File to copy: {01-copy_text.c}
```
Comment (RaduNichita) [inline Curs11/C/01-copy_text.c:56]: I don't think there are { and } in the output.

File: Curs11/C/01-copy_text.c [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+		Copy to: {test.c}
```
Comment (RaduNichita) [inline Curs11/C/01-copy_text.c:57]: you can add maybe the diff command?

File: Curs11/C/02-copy_binary.c [hunk 14]
```diff
@@ -0,0 +1,76 @@
+/* Programarea Calculatoarelor CA, 2022-2023
```
Comment (RaduNichita) [inline Curs11/C/02-copy_binary.c:1]: Please apply the comments from above also to this file.

File: Curs11/C/02-copy_binary.c [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+/*
+
+	compile: gcc 02-copy_binary.c -o copy_binary
+	run: ./copy_binary
+
+	Output:
+
+		File to copy: {a.exe}
```
Comment (RaduNichita) [inline Curs11/C/02-copy_binary.c:73]:
```suggestion
		File to copy: {a.bin}
```

**Review by RaduNichita — CHANGES_REQUESTED** [review]
Hi, @Grizzlly!

Thanks for this contribution, we really appreciate it. I left an initial round of comments, please solve them and will add more then.

Also, link the generic Makefile there, so everything should be one click away to run.

File: Curs11/C/01-copy_text.c (hunk 8 above)
Comment (Grizzlly) [reply Curs11/C/01-copy_text.c:16]: Why should I be using `fgets`?

File: Curs11/C/01-copy_text.c (hunk 7 above)
Comment (Grizzlly) [reply Curs11/C/01-copy_text.c:18]: Opening like this was a suggestion that @florinpop1981 made.

File: Curs09/C/02-max_struct.c [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define NAMES_COUNT 5
+#define MAX_NAME_LENGTH 8
+
+struct Student
+{
+	char Name[MAX_NAME_LENGTH];
```
Comment (RaduNichita) [inline Curs09/C/02-max_struct.c:17]: Please name the fields only with low letters.

File: Curs09/C/02-max_struct.c [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+	float Grade;
+};
+
+struct Student* get_random_students(const char names_array[][MAX_NAME_LENGTH])
```
Comment (RaduNichita) [inline Curs09/C/02-max_struct.c:21]: please find a proper name for this function. Something like `generating_random_grades`

File: Curs09/C/02-max_struct.c [hunk 18]
```diff
@@ -0,0 +1,81 @@
...
+{
+	struct Student* elevi = malloc(sizeof(struct Student) * NAMES_COUNT);
```
Comment (RaduNichita) [inline Curs09/C/02-max_struct.c:23]: please name all variables in English. Do not mix English with Romanian.

File: Curs09/C/02-max_struct.c [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+
+int main()
+{
+	const char names_array[NAMES_COUNT][MAX_NAME_LENGTH] = { 
+		"Popescu", "Stan", "Andrei", "Ion", "Darina" };
+
+	struct Student* students = get_random_students(names_array);
+	struct Student max_grade_student = students[0];
+	
+	printf("\nStudentii inregistrati:\n");
```
Comment (RaduNichita) [inline Curs09/C/02-max_struct.c:48]: Keep messages in english as well.

File: Curs11/C/01-copy_text.c [hunk 20]
```diff
@@ -0,0 +1,81 @@
...
+*	Program care realizeaza copierea unui fisier text.
+*
+*/
+
+#include <stdio.h>
+
+#define FILENAME_LENGTH 101
+
+/// <summary>
+/// Copies the contents from <paramref name="source" /> to <paramref name="dest" />.
```
Comment (RaduNichita) [inline Curs11/C/01-copy_text.c:12]: We don't do HTML here. 😅 

Maybe you can get inspired by Linux command line syntax from man.  [1]

[1] [man/cp](https://man7.org/linux/man-pages/man1/cp.1.html)

File: Curs11/C/01-copy_text.c (hunk 8 above)
Comment (RaduNichita) [reply Curs11/C/01-copy_text.c:41]: You can use also scanf, but remove the 100

File: Curs11/C/01-copy_text.c [hunk 21]
```diff
@@ -0,0 +1,81 @@
...
+int main()
+{
+	FILE *copy_from, *copy_to;
+	char filename[FILENAME_LENGTH];
+	int bytes_read = 0;
+
+	printf("File to copy: ");
+	scanf("%100s", filename);
+
+	if ((copy_from = fopen(filename, "r")) == NULL)
```
Comment (RaduNichita) [inline Curs11/C/01-copy_text.c:43]: Don't open files in if statements

File: Curs11/C/04-append_text.c [hunk 22]
```diff
@@ -0,0 +1,61 @@
...
+
+#include <stdio.h>
+#include <string.h>
+
+#define FILENAME_LENGTH 101
+#define TEXT_LENGTH 101
+
+int main()
+{
+	FILE *append;
```
Comment (RaduNichita) [inline Curs11/C/04-append_text.c:15]:
```suggestion
	FILE *append_file;
```

File: Curs11/C/04-append_text.c [hunk 23]
```diff
@@ -0,0 +1,61 @@
...
+	char filename[FILENAME_LENGTH], text[TEXT_LENGTH];
```
Comment (RaduNichita) [inline Curs11/C/04-append_text.c:16]:
```suggestion
	char filename[FILENAME_LENGTH], buffer[TEXT_LENGTH];
```

File: Curs11/C/04-append_text.c [hunk 24]
```diff
@@ -0,0 +1,61 @@
...
+*	compile: gcc 04-append_text.c -o append_text
+*	run: ./append_text
+*
+*	Input:
+*		File to append: 04-append_text.c
+*		Type text:
+*			ana_are_mere
+* 
+*	Output:
+*		Copied N bytes.
```
Comment (RaduNichita) [inline Curs11/C/04-append_text.c:59]:
```suggestion
*		Copied N bytes.
```

I think the output should say how many bytes were coppied

File: .gitignore [hunk 25]
```diff
@@ -0,0 +1,2 @@
+
+.vs/
```
Comment (RaduNichita) [inline .gitignore:2]:
```suggestion
.vs/

```

**Review by RaduNichita — CHANGES_REQUESTED** [review]
Please solve these. Then, link the Makefile to make sure the GitHub action still triggers for this folder.

Then, there will be another round of review. 

Also, don't forget to sign off your commits and do a rebase when you finish.

File: Curs11/C/01-copy_text.c (hunk 20 above)
Comment (Grizzlly) [reply Curs11/C/01-copy_text.c:12]: Sorry, this is how things are documented in Visual Studio. I will change it.

File: Curs11/C/01-copy_text.c (hunk 7 above)
Comment (Grizzlly) [reply Curs11/C/01-copy_text.c]: @RaduNichita

Comment (RaduNichita) [reply Curs11/C/01-copy_text.c]:
```cpp
FILE* copyFrom = fopen(filename, "r");
if (!copyFrom) {
    ...
}
```

**Pull request closed without merging**